shared/p2p/options.go

```diff
@@ -2,35 +2,82 @@ package p2p
 
 import (
 	"fmt"
+	"io/ioutil"
 	"net"
+	"os"
+
+	peer "github.com/libp2p/go-libp2p-peer"
 
 	"github.com/libp2p/go-libp2p"
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	"github.com/prysmaticlabs/prysm/shared/iputils"
+
 	filter "github.com/libp2p/go-maddr-filter"
 	ma "github.com/multiformats/go-multiaddr"
-	"github.com/prysmaticlabs/prysm/shared/iputils"
 )
 
 // buildOptions for the libp2p host.
 // TODO(287): Expand on these options and provide the option configuration via flags.
-// Currently, this is a random port and a (seemingly) consistent private key
-// identity.
-func buildOptions(cfg *ServerConfig) []libp2p.Option {
+func buildOptions(cfg *ServerConfig) ([]libp2p.Option, error) {
+	if cfg.PrvKey == "" {
+		cfg.PrvKey = defaultPrivateKeyFile
+	}
+	var prvKey crypto.PrivKey
+	if _, err := os.Stat(cfg.PrvKey); os.IsNotExist(err) {
```

Comment:
Please abstract all of this to its own function. `buildOptions` should be doing the bare minimum in terms of responsibility. These changes now make `buildOptions` responsible for reading files, etc. Please follow the pattern like `whitelistSubnet` or `optionConnectionManager`.